### PR TITLE
Update requirements of numpy 

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Updated to support version of numpy to 1.24,
- Updated requirements `numpy>=1.19.1, <1.24` to `numpy>=1.19.1, <1.25` 
- Remove using deprecated aliases https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations.
- Remove tests `mobilefacedet-v1-mxnet` model from mxnet, that does not support newer versions of numpy.

### Related tickets

115936

### CODE REVIEW

File: nncf/torch/utils.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def fp32_accum_wrapper(func):
     def wrapper(tensor_to_sum, ret_tensor):
         half = tensor_to_sum.dtype == np.float16
         if half:
-            tensor_to_sum = tensor_to_sum.astype(np.float)
+            tensor_to_sum = tensor_to_sum.astype(np.float64)
```
Comment (AlexKoff88) [inline nncf/torch/utils.py:135]: I think it should be cast to float32 if the input type if float32. The same logic as for float16.

File: setup.py [hunk 2]
```diff
@@ -105,7 +105,7 @@ def find_version(*file_paths):
     "texttable>=1.6.3",
     "scipy>=1.3.2, <1.11",
     "networkx>=2.6, <=2.8.2",  # see ticket 94048 or https://github.com/networkx/networkx/issues/5962
-    "numpy>=1.19.1, <1.24",
+    "numpy==1.24.4",
```
Comment (alexsu52) [inline setup.py:108]: Please, align numpy version with OpenVINO requirements `numpy>=1.19.1,<1.26`

**Review by alexsu52 — CHANGES_REQUESTED** [review]

**Comment by AlexanderDokuchaev** [comment]
run pytorch pre-commit tests

File: nncf/torch/utils.py (hunk 1 above)
Comment (AlexanderDokuchaev) [reply nncf/torch/utils.py]: Change to np.float32

**Review by alexsu52 — COMMENTED** [review]
please, provide torch e2e results for this PR

**Comment by AlexanderDokuchaev** [comment]
e2e/nightly/402 - :white_check_mark:

**Review by alexsu52 — APPROVED** [review]
LGTM

**Pull request merged**